### PR TITLE
Run IoT Edge Compatibility Script as Part of IoT Edge Check

### PR DESCRIPTION
**Pull request by nimanch**
To Enhance Platform Validation, Run IoT Edge Compatibility Script as part of our IoT Edge Checks. 

Testing Done
1. Built Local Packages as part of package.sh and ensure that for both rpm and deb packages -> compatibility script is part of the package ( Unpack rpm and debian packages manually)
2. Install deb package on Ubuntu 20.04 VM and verified that IoT Edge Compatibility Check Passes
image
3. Force Failure on Compatibility Script and Ensure Failure bubbles up
image

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: edgelet/build/linux/package.sh [hunk 1]
```diff
@@ -15,43 +15,43 @@ VERSION="${VERSION:-$DEFAULT_VERSION}"
 DOCKER_VOLUME_MOUNTS=''
```
Comment (nimanch) [inline edgelet/build/linux/package.sh:15]: No Changes to this file, Just auto formatting done.

File: edgelet/build/linux/package.sh [hunk 2]
```diff
@@ -60,37 +60,36 @@ if [ -z "$DOCKER_IMAGE" ]; then
...
 if [ -n "$RUST_TARGET" ]; then
     RUST_TARGET_COMMAND="rustup target add $RUST_TARGET &&"
 fi
 
-
 case "$PACKAGE_OS.$PACKAGE_ARCH" in
-    centos7.amd64)
-        SETUP_COMMAND=$'
+centos7.amd64)
+    SETUP_COMMAND=$'
```
Comment (damonbarry) [inline edgelet/build/linux/package.sh:82]: In general, it looks like the formatter aligned match arms with the outer case statement. But in situations like this one, where we use $-quoting, the string contents are left at their old indentation. I see why the formatter doesn't want to mess with string contents, but it looks weird. I'd suggest we either (a) find all the $-quoted command strings and back them up four spaces, or (b) don't format this file in this PR. I'd prefer (b), assuming it won't fail our checkin gate.

File: edgelet/contrib/debian/postinst [hunk 3]
```diff
@@ -5,74 +5,74 @@ USER=iotedge
...
+            mkdir -p /var/lib/aziot/edged
         fi
+        if [ ! -d /var/log/aziot/edged ]; then
+            mkdir -p /var/log/aziot/edged
+        fi
+        chown ${USER}:${GROUP} /var/lib/aziot/edged
+        chown ${USER}:${GROUP} /var/log/aziot/edged
+        chown root:root /etc/aziot/config.toml.edge.template
+        chown ${USER}:${GROUP} /etc/aziot/edged/config.toml.default
+        chown ${USER}:${GROUP} /etc/aziot/edged/aziot-compatibility.sh
```
[the remark this answers is not in the document]
Comment (damonbarry) [reply edgelet/contrib/debian/postinst:20]: If this was the only change, then I'd recommend we make only this change in this PR and do any major reformatting separately.

[the remark this answers is not in the document]
Comment (nimanch) [reply edgelet/contrib/debian/postinst]: Done

File: edgelet/build/linux/package.sh (hunk 2 above)
Comment (nimanch) [reply edgelet/build/linux/package.sh]: Removed changes to this file

**Review by damonbarry — APPROVED** [review]
What have you done to test this? Have you run the packages build and installed to make sure everything is where it should be?

**Comment by nimanch** [comment]
> What have you done to test this? Have you run the packages build and installed to make sure everything is where it should be?

@damonbarry : Sorry for not adding this earlier, I have added description of my tests in the PR Description. Let me know if that is sufficient

**Pull request merged**